ykrt/pass_finder/try_passes.py

```diff
@@ -66,53 +66,96 @@ def log(logf, s):
 def executable_exists(cmd):
     return shutil.which(cmd) is not None
 
-def get_all_passes(is_prelink):
-    cmd = get_opt_cmd(is_prelink) 
-    p = Popen(cmd, shell=True, stdout=PIPE, close_fds=True)
-    sout, serr = p.communicate() 
-    assert(p.returncode == 0)
+def split_passes(passes_string):
+    parts = []           
+    temp_part = []      
+    paren_stack = []     
+    angle_stack = []    
+
+    i = 0  
+    while i < len(passes_string):
+        char = passes_string[i]
+
+        if char == '(':
+            paren_stack.append(char)
+        elif char == ')':
+            if paren_stack:
```

Comment:
I don't think `paren_stack` can ever be non-true unless there's a bug. So I'd do:

```python
elif char == ')':
  paren_stack.pop()
```